src/modules/endpoint/DefaultDataContainer.js

```diff
@@ -167,35 +143,42 @@ class DefaultDataContainer implements DataContainer {
       }
     }
 
-    await this._databaseConnector.storeResourceCache(newResourceCache, this._context)
+    await this._databaseConnector.storeResourceCache(newResourceCache, context)
     this._resourceCache = newResourceCache
   }
 
-  setLastUpdate = async (lastUpdate: Moment) => {
-    if (this._context === null) {
-      throw Error('Context has not been set yet.')
-    }
-    await this._databaseConnector.storeLastUpdate(lastUpdate, this._context)
+  setLastUpdate = async (context: DatabaseContext, lastUpdate: Moment) => {
+    this.removeOutdated(context)
+
+    await this._databaseConnector.storeLastUpdate(lastUpdate, context)
     this._lastUpdate = lastUpdate
   }
 
-  categoriesAvailable (): boolean {
+  categoriesAvailable (context: DatabaseContext): boolean {
+    this.removeOutdated(context)
+
     return this._categoriesMap !== null
   }
 
-  languagesAvailable (): boolean {
+  languagesAvailable (city: string): boolean {
```

Comment:
Why do we need the city here? I think you mean something like
```
return this.getLanguagesForCity(city) !== null
```
(with a proper function `getLanguagesForCity` of course)

Reply:
Languages depend on the city. This is needed now.